relayer/Dockerfile

```diff
@@ -0,0 +1,6 @@
+FROM debian:11-slim
```

Comment:
Compare to the old `repo_root/Dockerfile`. I guess I made too many changes for the PR to reflect that the file was just renamed and edited.

The edits I made were to change the base image and to drop the build `ARG`. The base image was formerly `golang`, but we're not actually building any Go code in the building of this image (rather, goreleaser is pre-building the binary, and then the Dockerfile just copies that binary into the image) so there's no need to have a Go build environment. Furthermore, there's no reason to pass in a `GO_VERSION` either.